File: 2016/src/day2.py
inputs = open("../inputs/day2.txt", "r").read().split("\n")
inputs = [i for i in inputs if len(i) > 0]

x, y = 0, 2
grid = [
    [0,  0,  1,   0,  0],
    [0,  2,  3,   4,  0],
    [5,  6,  7,   8,  9],
    [0, "A", "B","C", 0],
    [0,  0,  "D", 0,  0]
]

code = ""

for line in inputs:
    for i in line:
        if i == "L":
            x = x - 1 if (x - 1 )>= 0 and grid[y][x - 1] != 0 else x
        elif i == "R":
            x = x + 1 if (x + 1) <= 4 and grid[y][x + 1] != 0 else x
        elif i == "U":
            y = y - 1 if (y - 1) >= 0 and grid[y - 1][x] != 0 else y
        else:
            y = y + 1 if (y + 1) <= 4 and grid[y + 1][x] != 0 else y
    
    code += str(grid[y][x])

print(code)

File: 2016/day2.rs
use std::fs;

fn main() {
    let inputs = fs::read_to_string("../inputs/day2.txt").expect("Error when reading file."); 
    let inputs_split = inputs.split("\r\n");
    let commands = inputs_split.collect::<Vec<&str>>();

    let mut x: i32 = 1;
    let mut y: i32 = 1;

    let grid: [[i32; 3]; 3] = [
        [ 1, 2, 3 ],
        [ 4, 5, 6 ],
        [ 7, 8, 9 ]
    ];

    for line in commands {
        for c in line.chars() {
            match c {
                'U' => y = if (y - 1) > 0 {y - 1} else {y},
                'D' => y = if (y + 1) < 2 {y + 1} else {y},
                'R' => x = if (x + 1) < 2 {x + 1} else {x},
                'L' => x = if (x - 1) > 0 {x - 1} else {x},
                _   => todo(),
            }
        }
        println!("{}, {}, {}", x, y, grid[y][x]);
    }   
}
